conntrack/conntrack.go

```diff
@@ -22,11 +22,20 @@ import (
 	log "github.com/Sirupsen/logrus"
 )
 
-var conntrackDirections = []string{
+// For TCP/UDP, each conntrack entry holds two copies of the tuple
+// (src addr, dst addr, src port, dst port).  One copy for the original direction and one copy for
+// the reply direction.  This is how the kernel handles NAT: by looking up the tuple for a packet
+// by its original tuple and mapping onto the corresponding reply direction tuple (or vice versa).
+//
+// When we delete conntrack entries by IP address, we need to specify which element of the tuple
+// to look in.  This slice holds the flags corresponding to the fields we care about.  Since we're
+// deleting entries for local workload endpoints, either the endpoint originated the traffic, or it
+// received the traffic and replied to it.  In the originating case, the "original source" will be
+// set to the endpoint's IP; in the other case, the "reply source". Hence, it's sufficient to only
+// look in those two fields.
```

Comment:
This comment is very helpful, but I think it may need even a bit more in order to explain why only looking at orig-src and reply-src is watertight for our purposes.

The concerns are, I think that:
- There could be a (partial) flow with `orig-dst==<endpoint IP>` but no `reply-src`, if the endpoint chooses not to respond at all.  At least with UDP that would be quite easy to arrange; probably more tricky with TCP.
- Is there any way to get a flow with `reply-dst==<endpoint IP>!=orig-src` ?

I think the possible answers are that:
- It isn't possible to get conntrack state at all for such cases, because they fall outside the definition of what conntrack considers to be a flow; and/or
- There can be conntrack state for such cases, but it naturally expires so quickly that we don't care about Felix deleting it here.  (Hypothesis:) Conntrack state is only long-lived for complete flows with `reply-src==orig-dst` and `reply-dst==orig-src`, and it's only that long-lived state that we are concerned about deleting here.

Feel free to cut and paste if any of this is right or useful.